internal/factsengine/gatherers/saphostctrl_test.go

```diff
@@ -0,0 +1,198 @@
+					"changelist": &entities.FactValueString{Value: "2091708"},
+					"hostname":   &entities.FactValueString{Value: "s41app"},
+					"instance":   &entities.FactValueString{Value: "40"},
+					"patch":      &entities.FactValueString{Value: "50"},
+					"revision":   &entities.FactValueString{Value: "785"},
+					"system":     &entities.FactValueString{Value: "S41"},
+				}},
+				&entities.FactValueMap{Value: map[string]entities.FactValue{
+					"changelist": &entities.FactValueString{Value: "2069355"},
+					"hostname":   &entities.FactValueString{Value: "s41db"},
+					"instance":   &entities.FactValueString{Value: "11"},
+					"patch":      &entities.FactValueString{Value: "819"},
+					"revision":   &entities.FactValueString{Value: "753"},
+					"system":     &entities.FactValueString{Value: "HS1"},
+				}},
+			}},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SapHostCtrlTestSuite) TestSapHostCtrlGatherPingSuccess() {
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "Ping").Return(
+		[]byte("SUCCESS (  543341 usec)\n"), nil)
+
+	p := gatherers.NewSapHostCtrlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "ping",
+			Gatherer: "saphostctrl",
+			Argument: "Ping",
+			CheckID:  "check1",
+		},
+	}
+
+	factResults, err := p.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:    "ping",
+			CheckID: "check1",
+			Value: &entities.FactValueMap{
+				Value: map[string]entities.FactValue{
+					"status":  &entities.FactValueString{Value: "SUCCESS"},
+					"elapsed": &entities.FactValueString{Value: "543341"},
+				},
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SapHostCtrlTestSuite) TestSapHostCtrlGatherPingFailed() {
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "Ping").Return(
+		[]byte("FAILED (     497 usec)\n"), nil)
+
+	p := gatherers.NewSapHostCtrlGatherer(suite.mockExecutor)
+
+	factRequests := []entities.FactRequest{
+		{
+			Name:     "ping",
+			Gatherer: "saphostctrl",
+			Argument: "Ping",
+			CheckID:  "check1",
+		},
+	}
+
+	factResults, err := p.Gather(factRequests)
+
+	expectedResults := []entities.Fact{
+		{
+			Name:    "ping",
+			CheckID: "check1",
+			Value: &entities.FactValueMap{
+				Value: map[string]entities.FactValue{
+					"status":  &entities.FactValueString{Value: "FAILED"},
+					"elapsed": &entities.FactValueString{Value: "497"},
+				},
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.ElementsMatch(expectedResults, factResults)
+}
+
+func (suite *SapHostCtrlTestSuite) TestSapHostCtrlGatherError() {
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "Ping").Return(
+		[]byte("FAILURE\n"), nil)
+	suite.mockExecutor.On("Exec", "/usr/sap/hostctrl/exe/saphostctrl", "-function", "Pong").Return(
```

Comment:
What about an scenario where the command fails.
Here we have a
- parsing error
- unsupported command

So we miss the command error test